lib/plug/conn/query.ex

```diff
@@ -63,7 +63,10 @@ defmodule Plug.Conn.Query do
   end
 
   def decode(query, initial) do
-    parts = :binary.split(query, "&", [:global])
+    parts = query
+    |> String.trim_trailing("&")
```

Comment:
Rather than trimming the trailing, we should probably just ignore any pair that is an empty string on `decode_srtring_pair`, as that would also handle input such as "foo=bar&&baz=bat". Could you please investigate?